utils/bundle-sim/src/bin/bundle-sim.rs

```diff
@@ -0,0 +1,242 @@
+            }
+            etcd.put(key.clone(), value, None).await?;
+        }
+        Ok(())
+    }
+    fn cluster_info(&self, etcd_dump: &std::fs::File) -> anyhow::Result<(String, String)> {
+        use std::io::{BufRead, BufReader};
+        let mut reader = BufReader::new(etcd_dump);
+        let mut first_line = String::new();
+        reader.read_line(&mut first_line)?;
+
+        let parts: Vec<&str> = first_line.split('/').collect();
+        let cluster_id = parts
+            .iter()
+            .position(|&x| x == "clusters")
+            .and_then(|i| parts.get(i + 1))
+            .ok_or(anyhow::anyhow!(
+                "No cluster id in the first etcd_dump line: {first_line}"
+            ))?;
+        let cluster_ns = parts
+            .iter()
+            .position(|&x| x == "namespaces")
+            .and_then(|i| parts.get(i + 1))
+            .ok_or(anyhow::anyhow!(
+                "No cluster namespace in the first etcd_dump line: {first_line}"
+            ))?;
+        Ok((cluster_id.to_string(), cluster_ns.to_string()))
+    }
+    fn extract(&mut self) -> anyhow::Result<()> {
+        println!();
+        anyhow::ensure!(
+            self.bundle.exists(),
+            "--bundle doesn't point to an existing file/dir"
+        );
+        if self.bundle.is_file() {
+            let file = std::fs::File::open(&self.bundle)?;
+            std::fs::create_dir_all(&self.untar_path)?;
+
+            let gz_decoder = flate2::read::GzDecoder::new(file);
+            let mut archive = tar::Archive::new(gz_decoder);
+
+            // Define the directories you want to extract
+            let target_dirs = vec![
+                "topology/node",
+                "topology/pool",
+                "topology/volume",
+                "etcd_dump",
+            ];
+            let mut unpacked_dirs = HashSet::<&str>::new();
+
+            for entry in archive.entries()? {
+                let mut entry = entry?;
+                let path = entry.path()?;
+
+                // Check if the path starts with any of the target directories
+                if let Some(target_dir) = target_dirs.iter().find(|dir| path.starts_with(dir)) {
+                    tracing::trace!("Extracting {}", path.display());
+                    if !unpacked_dirs.contains(target_dir) {
+                        tracing::info!("Extracting {target_dir}");
+                    }
+                    entry.unpack_in(&self.untar_path)?;
+
+                    unpacked_dirs.insert(target_dir);
+                }
+            }
+
+            if unpacked_dirs.len() != target_dirs.len() {
+                let mut target_dirs = target_dirs;
+                target_dirs.retain(|t| !unpacked_dirs.contains(t));
+                anyhow::bail!("{target_dirs:?} were not found in the bundle file")
+            }
+            self.bundle = self.untar_path.clone();
+        } else {
+            anyhow::ensure!(self.bundle.is_dir(), "--bundle must be a directory");
```

Comment:
```suggestion
            anyhow::ensure!(self.bundle.is_dir(), "--bundle arg must be a file/directory");
```

Reply:
indeed, I moved the ensure to the one at the start of the function